src/libraries/System.Collections.Immutable/tests/ImmutableDictionaryTest.cs

```diff
@@ -24,6 +24,18 @@ public void AddExistingKeyDifferentValueTest()
             AddExistingKeyDifferentValueTestHelper(Empty(StringComparer.Ordinal, StringComparer.Ordinal), "Company", "Microsoft", "MICROSOFT");
         }
 
+        [Fact]
+        public void AddRangeShouldThrowOnNullKeyTest()
+        {
+            var items = new[] { new KeyValuePair<string, string>(null, "value") };
+
+            var map = Empty(StringComparer.Ordinal, StringComparer.Ordinal);
+            Assert.Throws<ArgumentNullException>(() => map.AddRange(items));
```

Comment:
The same issue impacts the `SetRange` method and the corresponding methods in the builder type. Would it be possible to add tests for those as well? Thanks.

Reply:
@eiriktsarpalis I added tests for SetItems and AddRange for the builder type. I could add the similar test for the (unaffected) classes ImmutableSortedDictionary and ImmutableHashSet if you think that it's useful.

Reply:
I think this is fine 👍